### ISSUE TITLE
Convolution with no bias

### ISSUE THREAD

**Issue by bearpaw** [viorik/ConvLSTM#16]
Hi. You use the convolution with NO bias in ConvLSTM.lua, and the normal convolution with bias in UntiedConvLSTM.lua. Could you please explain about what's the difference with or without bias? Thank you!

**Comment by viorik** [comment]
Hi @bearpaw 
ConvLSTM actually uses both SpatialConvolution and SpatialConvolutionNoBias, whereas Untied uses only SpatialConvolution.
The explanation: As common with LSTMs, in ConvLSTM the activation of each gate is obtained by performing two operations (here convolutions), one on the input (x_t) and one on the previous state (h_{t-1}), and eventually adding **one** bias -- check equation 1 in [the paper](http://arxiv.org/pdf/1511.06309v5.pdf). Now, the classic SpatialConvolution layer implements `w*x + b`, i.e. convolution plus bias. If you would use two classic SpatialConvolution layers inside the gates in LSTM you would actually end up with 2 biases per feature map. Which is not a big issue, but it is an unnecessary increase in the number of parameters of the model. Hence ConvolutionNoBias layer only eliminates the bias, performing only `w*x`. However, note that more recently, the SpatialConvolution in torch provides the option of setting bias to nil by calling noBias(). This has the same effect as using ConvolutionNoBias layer.
Now, UntiedConvLSTM IS ConvLSTM, but with the first step using a simpler activation model. Since at the first step, there is no previous state (h_{t-1} does not exist), we only need to convolve the input and add one bias, hence use one classic SpatialConvolution. 

Hope this helps. Note that the classic [LSTM in rnn](https://github.com/Element-Research/rnn/blob/master/LSTM.lua) from where my implementation is derived, uses similarly Linear and LinearNoBias layers.

**Comment by bearpaw** [comment]
Hi @viorik .
Thanks for your explanation! It's very comprehensive! 

Currently, I'm using ConvLSTM in my own project. When the kernelSize and kernelSizeMemory are both set as 1, the training goes well. However, when I increase the kernelSize, the network suffers severe overfitting (high training accuracy but very low validation accuracy). Do you have any suggestions about this issue? For example, is it possible to add dropout in LSTM (I'm not sure whether this is theoretically correct )?

**Comment by viorik** [comment]
What problem are you trying to solve? Maybe you don't have enough training data and issues appear when you increase the size of the model. I think dropout could be used, but haven't tried myself.

**Comment by bearpaw** [comment]
@viorik Got it. I will have a try. Thank you!
